src/cli/cli_joiner.cpp

```diff
@@ -43,10 +45,55 @@ namespace ot {
 namespace Cli {
 
 const struct Joiner::Command Joiner::sCommands[] = {
-    {"discerner", &Joiner::ProcessDiscerner}, {"help", &Joiner::ProcessHelp}, {"id", &Joiner::ProcessId},
-    {"start", &Joiner::ProcessStart},         {"stop", &Joiner::ProcessStop},
+    {"advertisement", &Joiner::ProcessAdvertisement},
+    {"discerner", &Joiner::ProcessDiscerner},
+    {"help", &Joiner::ProcessHelp},
+    {"id", &Joiner::ProcessId},
+    {"start", &Joiner::ProcessStart},
+    {"stop", &Joiner::ProcessStop},
 };
 
+otError Joiner::ProcessAdvertisement(uint8_t aArgsLength, char *aArgs[])
+{
+    otError error = OT_ERROR_NONE;
+
+    if (aArgsLength == 1)
+    {
+        uint32_t       oui;
+        uint8_t        advDatalength;
+        const uint8_t *advData;
+
+        advData = otThreadGetJoinerAdvertisement(mInterpreter.mInstance, &oui, &advDatalength);
+        if (advData != nullptr)
+        {
+            mInterpreter.OutputFormat("oui: 0x%06x\r\n", oui);
+            mInterpreter.OutputFormat("advData: ");
+            mInterpreter.OutputBytes(advData, advDatalength);
+            mInterpreter.OutputFormat("\r\n");
```

Comment:
The `Output` method requires an additional `aBufLength` arguments. it requires counting string length by the caller, so I choose to stick with `OutputFormat` as it is simpler.

Reply:
Worth to write a new version of `Output(const char *s)` ?

Reply:
Maybe, but better not in this PR. Because there are many other usages of `OutputFormat` with the same cause, and it is better to have they all using the new version of `Output` once at all. So I think it is better to do this in a separate PR,  Agree?

Reply:
agree